imap_processing/glows/l0/glows_l0_data.py

```diff
@@ -0,0 +1,153 @@
+class GlowsHistL0:
+    """Data structure for storing GLOWS histogram packet data.
+    Parameters
+    ----------
+    packet : tuple[list]
+        Histogram packet yielded from space_packet_parser.generate_packets
+
+    Attributes
+    ----------
+    L0_KEYS : tuple[str]
+        Data names from the packet
+    data: dict
+        The data from the packet
+
+    Methods
+    -------
+    process_packet(packet)
+        Process the given packet into the data dictionary
+
+    _convert_histogram_data(binary_hist_data)
+        Process the binary histogram data into 8-bit bins, in a list.
+    """
+
+    L0_KEYS: tuple[str] = (
+        "SHCOARSE",
+        "STARTID",
+        "ENDID",
+        "FLAGS",
+        "SWVER",
+        "SEC",
+        "SUBSEC",
+        "OFFSETSEC",
+        "OFFSETSUBSEC",
+        "GLXSEC",
+        "GLXSUBSEC",
+        "GLXOFFSEC",
+        "GLXOFFSUBSEC",
+        "SPINS",
+        "NBINS",
+        "TEMPAVG",
+        "TEMPVAR",
+        "HVAVG",
+        "HVVAR",
+        "SPAVG",
+        "SPVAR",
+        "ELAVG",
+        "ELVAR",
+        "EVENTS",
+        "HISTOGRAM_DATA",
+    )
+
+    def __repr__(self):
+        return str(self.data)
+
+    def __init__(self, packet):
+        self.data = self.process_packet(packet)
+
+    def process_packet(self, packet) -> dict:
+        """Process the GLOWS histogram packet, which is yielded from space_packet_parser
+
+        Parameters
+        ----------
+        packet
+            Histogram packet from space_packet_parser
+        Returns
+        -------
+        data: dict
+            Data dictionary created from the packet, using L0_KEYS
+        """
+        data = {}
+        for key in self.L0_KEYS:
+            if key != "HISTOGRAM_DATA":
+                data[key] = packet.data[key].derived_value
+            else:
+                data[key] = self._convert_histogram_data(packet.data[key].raw_value)
+            if data[key] is None:
+                raise ValueError(f"Missing data for {key} in {packet.data}")
```

Comment:
Is this meaning that the packet definition isn't filled out properly? The derived_value returning None seems a bit concerning. I figured it would have been converted to int/float according to your definition in the xml.

Reply:
It's not supposed to happen, that's why it throws an error. I came across it once when I had a bad XTCE so I figured I should throw up an alert just in case rather than having some other random piece fail due to missing data.